src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/compute/v5_4/OpenNebulaComputePluginTest.java

```diff
@@ -97,761 +97,441 @@
-
-		Assert.assertEquals(expected, this.plugin.getFlavors().size());
-	}
-	
-	// test case: When calling the updateHardwareRequirements method and getting
-	// invalid resource values in the cloud, it must not add them to the existing
-	// flavor set.
+	// test case: when invoking the getCreateRequestInstance method with a valid client and compute order,
+	// the plugin should create the base64 script using the appropriate launch command generator; create
+	// the network ids list; and, find an appropriate 'flavor' for the order.
 	@Test
-	public void testUpdateHardwareRequirementsWithInvalidResourceValue() throws UnexpectedException {
+	public void testGetCreateRequestInstance() throws UnexpectedException, NoAvailableResourcesException {
 		// set up
-		ImagePool imagePool = Mockito.mock(ImagePool.class);
-		PowerMockito.mockStatic(OpenNebulaClientUtil.class);
-		PowerMockito.when(OpenNebulaClientUtil.getImagePool(Mockito.any(Client.class))).thenReturn(imagePool);
-
-		Image image = Mockito.mock(Image.class);
-		Iterator<Image> imageIterator = Mockito.mock(Iterator.class);
-		Mockito.when(imageIterator.hasNext()).thenReturn(true, false);
-		Mockito.when(imageIterator.next()).thenReturn(image);
-		Mockito.when(imagePool.iterator()).thenReturn(imageIterator);
-		Mockito.when(image.xpath(IMAGE_SIZE_PATH)).thenReturn(IMAGE_SIZE_STRING_VALUE);
-
-		TemplatePool templatePool = Mockito.mock(TemplatePool.class);
-		PowerMockito.when(OpenNebulaClientUtil.getTemplatePool(Mockito.any(Client.class))).thenReturn(templatePool);
-
-		Template template = Mockito.mock(Template.class);
-		Iterator<Template> templateIterator = Mockito.mock(Iterator.class);
-		Mockito.when(templateIterator.hasNext()).thenReturn(true, false);
-		Mockito.when(templateIterator.next()).thenReturn(template);
-		Mockito.when(templatePool.iterator()).thenReturn(templateIterator);
-		Mockito.when(template.xpath(TEMPLATE_CPU_PATH)).thenReturn(DECIMAL_STRING_VALUE);
-		Mockito.when(template.xpath(TEMPLATE_MEMORY_PATH)).thenReturn(DECIMAL_STRING_VALUE);
-		Mockito.when(template.xpath(TEMPLATE_IMAGE_ID_PATH)).thenReturn(DECIMAL_STRING_VALUE);
-
-		Mockito.doReturn(0).when(this.plugin).getDiskSizeFromImageSizeMap(Mockito.anyMap(), Mockito.anyString());
-
-		TreeSet<HardwareRequirements> expected = new TreeSet<HardwareRequirements>();
-
-		Client client = Mockito.mock(Client.class);
+		LaunchCommandGenerator launchCommandGenerator = Mockito.mock(LaunchCommandGenerator.class);
+		this.plugin.setLaunchCommandGenerator(launchCommandGenerator);
+
+		Mockito.when(launchCommandGenerator.createLaunchCommand(Mockito.any(ComputeOrder.class))).thenReturn(FAKE_BASE64_SCRIPT);
+		Mockito.doReturn(this.networkIds).when(this.plugin).getNetworkIds(Mockito.anyList());
+		Mockito.doReturn(this.hardwareRequirements).when(this.plugin).findSmallestFlavor(
+				Mockito.any(Client.class), Mockito.any(ComputeOrder.class));
+		Mockito.doReturn(this.getCreateComputeRequest()).when(this.plugin).createComputeRequest(
```

Comment:
I believe in an attempt to decrease the size of the _getCreateRequestInstance_ method in the plugin, the entire function of the **Builder** pattern was eliminated by creating the _createComputeRequest_ method with thirteen-parameter.

I recommend to bring the building of the **CreateComputeRequest** object to the _getCreateComputeRequest_ method and renaming it to _doCreateComputeRequest_.

Reply:
done